tests/artifacts_tests/test_download_artifact.py

```diff
@@ -0,0 +1,41 @@
+from __future__ import annotations
+
+import pathlib
+
+import pytest
+
+import optuna
+from optuna.artifacts import FileSystemArtifactStore
+from optuna.artifacts._download import download_artifact
+from optuna.artifacts._protocol import ArtifactStore
+from optuna.artifacts._upload import upload_artifact
```

Comment:
Generally, we test public functions/classes instead of private ones if possible (c.f., [Optuna's test policy](https://github.com/optuna/optuna/wiki/Test-Policy#code-quality)). How about importing these from `optuna.artifacts`, excluding the private ones?